src/app/layout.tsx

```diff
@@ -7,16 +7,45 @@ import '@patternfly/react-styles/css/components/Menu/menu.css';
 
 export const metadata = {
   title: 'InstructLab UI',
-  description: 'InstructLab UI'
+  description: 'InstructLab UI',
 };
 
 interface RootLayoutProps {
   children: ReactNode;
 }
 
 const RootLayout = ({ children }: RootLayoutProps) => {
+  // Determine the environment based on hostname
```

Comment:
I think rather than detecting whether it's prod or qa deployment in the code, we can simply provide the environment variables that takes the hostname, scriptsource and websiteId as an variable in .env. In the code it just use those values, irrespective of prod or qa deployment. Because these are basically day 0 configuration variables that user need to configure before the deployment.

Reply:
Right so seems I forgot that we want to support local deployments with users injecting their own script tags - so we will need some env variable to track those for website-code and for script source. I dont think we want to set the Prod and QA values to be configurable though, otherwise we could get someone forwarding metrics from their local deployment to either one, so I think we should keep those checks  in the source, and then add an option if its neither to check if the custom script tag and website-code are defined, and if so render those.

Reply:
the challenge is, these URL can possibly change, for example we are planning to migrate `ui.instructlab.ai` to `contribute.instructlab.ai`. If that happens, we will have to make the code level changes to fix the umami metrics. 

How about we have a environment variable `DEPLOYMENT_ENVIRONMENT` that takes two value `prod` and `qa`, and use that to detect the deployment environment. If you see concern with that, we can go with checking the url itself.

Reply:
You got it, lets try it that way first and see what we think.